tests/test_flow/test_outputs.py

```diff
@@ -123,10 +123,21 @@ def three_outputs():
         flow.get("a")
 
 
+def test_non_sequence_output(builder):
+    @builder
+    @bn.outputs("a")
+    def non_sequence_output():
+        return 7
+
+    flow = builder.build()
+    with pytest.raises(EntityValueError):
```

Comment:
Since we're changing the error message here, we might want to check that the exception has (part of) the message we expect. (You can see an example of this in `tests/test_flow/test_api.py::test_get_modes_not_persisted`.)